extras/asr-services/providers/vibevoice/service.py

```
@@ -102,6 +102,24 @@ def get_capabilities(self) -> list[str]:
+        if self.transcriber is None:
+            return False
+        return self.transcriber.supports_batch_progress(audio_duration)
+
+    def transcribe_with_progress(self, audio_file_path: str, context_info=None):
+        """Yield progress counters then final result for long audio.
+
+        Delegates to the transcriber's _transcribe_batched_with_progress() generator.
+        Runs synchronously (called via run_in_executor by the endpoint).
+        """
+        if self.transcriber is None:
+            raise RuntimeError("Service not initialized")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)